src/industry_cmd.cpp

```diff
@@ -1218,6 +1218,23 @@ static CommandCost CheckNewIndustry_Forest(TileIndex tile)
 	return CommandCost();
 }
 
+static bool CheckScaledDistanceFromEdge(TileIndex tile, uint maxdist)
+{
+	uint maxdist_x = maxdist;
+	uint maxdist_y = maxdist;
+
+	if (MapSizeX() > 256) maxdist_x *= MapSizeX() / 256;
+	if (MapSizeY() > 256) maxdist_y *= MapSizeY() / 256;
+
+	bool ok = false;
+	ok |= DistanceFromEdgeDir(tile, DIAGDIR_NE) < maxdist_x;
```

Comment:
no real reason not to use `||=` here, and it'll shortcircuit

Reply:
huh, could've sworn `||=` was a thing. Still, `ok = ok || ...` ?

Reply:
May as well just make a series of `if (distance < max) return true;` lines instead of keeping a variable.